fastai/learner.py

```diff
@@ -97,22 +97,22 @@ def before_epoch(self):
 class Learner(GetAttr):
     _default='model'
     def __init__(self,
-                 dls,  # `DataLoaders` containing data for each dataset needed for `model`
-                 model:callable, #  The model to train or use for inference
-                 loss_func:callable|None=None,  # Loss function for training
-                 opt_func=Adam,  # Optimisation function for training
-                 lr=defaults.lr,  # Learning rate
-                 splitter:callable=trainable_params,  # Used to split parameters into layer groups
-                 cbs=None,  # Callbacks
-                 metrics=None,  # Printed after each epoch
-                 path=None,  # Parent directory to save, load, and export models
-                 model_dir='models',  # Subdirectory to save and load models
-                 wd=None,  # Weight decay
-                 wd_bn_bias=False,  # Apply weight decay to batchnorm bias params?
-                 train_bn=True,  # Always train batchnorm layers?
-                 moms=(0.95,0.85,0.95),  # Momentum
-                 default_cbs:bool=True  # Include default callbacks?
-                ):
+        dls:DataLoaders, # `DataLoaders` containing fastai or PyTorch `DataLoader`s
+        model:callable|nn.Module, # PyTorch model for training or inference
+        loss_func:nn.Module|None=None, # Loss function. Defaults to `dls` loss
+        opt_func:Optimizer|OptimWrapper=Adam, # Optimization function for training
+        lr:float|slice=defaults.lr, # Default learning rate
+        splitter:callable=trainable_params, # Split model into parameter groups. Defaults to one parameter group
+        cbs:Callback|list|None=None, # `Callback`s to add to `Learner`
```

Comment:
I think any collection works - doesn't have to be a `list`.

Reply:
The convention of `Object|list` is incorrectly used all over the codebase to represent a collection/iterable of the first object following the docment sprint. I kept with the current codebase since we don't have a defined method of any iterable/collection. What should it be changed to?

Reply:
Thanks for letting me know - I think we should change `list` to `collections.abc.MutableSequence` when it's used in this way.

Reply:
Changing all of them is outside the scope of this PR. I can change these few instances to `cbs:Callback|MutableSequence[Callback]|None`, the less technically correct `cbs:Callback|MutableSequence|None`, or leave it as `Callback|list` to match the rest of the library.

Alternatively, before merging #3639 with the current `Object|list` convention, you came up with the custom type hint of [`Listified[Object]`](https://github.com/fastaidocsprint/styleguide#custom-type-aliases) to represent `Object|MutableSequence[Object]`. If you prefer, I could use that instead. And perhaps add `MutableSequence` to the custom Listified type hint.

Reply:
We can't use `Listified[Object]` since it's a generic type. But we can use `Listified`. Since `list` is a `MutableSequence` we can simply change the `Listified` type -- no need to add to it.

Reply:
I'm still not completely sure what you want the type hint to be changed too.

Reply:
`Callback|MutableSequence|None`